### PR TITLE
Fixes issue with headers not being a dict in AWS lambda instr

### PR DESCRIPTION
**Pull request by kubawach**
# Description

Fixes issue with AWS lambda - in some rare cases (API GW proxy integration test) `headers` are set to None, breaking context propagators.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- manual testing (pretty straightforward change)

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py [hunk 1]
```diff
@@ -108,6 +108,8 @@ def _default_event_context_extractor(lambda_event: Any) -> Context:
     Assumes the Lambda Event is a map with the headers under the 'headers' key.
     This is the mapping to use when the Lambda is invoked by an API Gateway
     REST API where API Gateway is acting as a pure proxy for the request.
+    Protects headers from being something other than dictionary, as this
+    is what downstream propagators
```
Comment (ocelotl) [inline instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py:112]: Is something missing from this sentence?

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,16 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [1.11.0-0.30b0](https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.11.0-0.30b0) - 2022-04-18
 
 ### Fixed
+- `opentelemetry-instrumentation-aws-lambda` Fixed an issue - in some rare cases (API GW proxy integration test)
```
Comment (srikanthccv) [inline CHANGELOG.md:13]: Please move this to Unreleased

**Review by sanketmehta28 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (srikanthccv) [reply CHANGELOG.md:18]: @kubawach Please move this to unreleased section.

Comment (kubawach) [reply CHANGELOG.md:11]: Done (for the second time already ;) ).

**Comment by kubawach** [comment]
TY @sanketmehta28 ! :)

**Pull request merged**